packages/nextra/src/mdx-plugins/remark.ts

```diff
@@ -45,11 +46,20 @@ export function remarkHeadings(this: Processor) {
       },
       node => {
         if (node.type === 'heading') {
+          const jsxH1 =
+            node.depth === 1 &&
+            Array.isArray(node.children) &&
+            node.children.some(
+              (child: any) => (child.type as string) === 'mdxJsxTextElement'
+            )
           const heading = {
             ...(node as Heading),
             value: getFlattenedValue(node)
           }
           data.headingMeta.headings.push(heading)
+          if (jsxH1) {
+            data.headingMeta.jsxH1 = true
```

Comment:
could you also add a comment on why this prop is needed?

Reply:
```mdx
import { useRouter } from 'next/router'

export const TagName = () => {
  const { tag } = useRouter().query
  return tag || null
}

# Posts Tagged with “<TagName/>”
```

For headings contains jsx